MobileCenterPush/MobileCenterPushTests/MobileCenterPushTests.m

```diff
@@ -0,0 +1,139 @@
+#import <Foundation/Foundation.h>
+#import <XCTest/XCTest.h>
+#import <OCMock/OCMock.h>
+#import <OCHamcrestIOS/OCHamcrestIOS.h>
+
+#import "MSService.h"
+#import "MSServiceAbstract.h"
+#import "MSServiceInternal.h"
+
+#import "MSPush.h"
+#import "MSPushPrivate.h"
+#import "MSPushInternal.h"
+#import "MSMockPushDelegate.h"
+#import "MSPushLog.h"
+
+static NSString *const kMSTestAppSecret = @"TestAppSecret";
+static NSString *const kMSTestDeviceToken = @"TestDeviceToken";
+
+@interface MSPushTests : XCTestCase <MSPushDelegate>
```

Comment:
If you need delegates, please instantiate in local and use it for each method.
This test class will grow and need to verify most of push functionalities. I don't think we will have same delegate for all test cases.